src/adapter/src/coord.rs

```diff
@@ -791,6 +792,38 @@ impl<S: Append + 'static> Coordinator<S> {
             }
         }
 
+        // Migrate builtin objects
+        for (compute_id, sink_ids) in builtin_migration_metadata.previous_sink_ids {
+            self.controller
+                .compute_mut(compute_id)
+                .unwrap()
+                .drop_sinks_unvalidated(sink_ids)
+                .await?;
+        }
+        for (compute_id, index_ids) in builtin_migration_metadata.previous_index_ids {
+            self.controller
+                .compute_mut(compute_id)
+                .unwrap()
+                .drop_indexes_unvalidated(index_ids)
+                .await?;
+        }
+        for (compute_id, recorded_view_ids) in builtin_migration_metadata.previous_recorded_view_ids
+        {
+            self.controller
+                .compute_mut(compute_id)
+                .unwrap()
+                .drop_sinks_unvalidated(recorded_view_ids.clone())
+                .await?;
+            self.controller
+                .storage_mut()
+                .drop_sources_unvalidated(recorded_view_ids)
+                .await?;
+        }
+        self.controller
+            .storage_mut()
+            .drop_sources_unvalidated(builtin_migration_metadata.previous_source_ids)
+            .await?;
+
```

Comment:
I'm not really a big fan of all these "unvalidated" methods, but I'm not sure what else to do.